src/ucp/api/ucp.h

```diff
@@ -2152,8 +2152,11 @@ ucs_status_t ucp_rkey_ptr(ucp_rkey_h rkey, uint64_t raddr, void **addr_p);
  * @li Once the RKEY object is released an access to the memory will cause an
  * undefined failure.
  * @li If the RKEY object was not created using
- * @ref ucp_ep_rkey_unpack "ucp_ep_rkey_unpack()" routine the behaviour of this
+ * @ref ucp_ep_rkey_unpack "ucp_ep_rkey_unpack()" routine the behavior of this
  * routine is undefined.
+ * @li The RKEY object must be destroyed after all outstanding operations which
+ * are using it are flushed, and before the endpoint on which it was unpacked
+ * is destroyed.
```

Comment:
Do we manage some sort ref counting for keys ?

Reply:
no. what counting did you have in mind?

Reply:
I was thinking to manager reference counter on the object, so whenever you try to release rkey while still having outstanding request it will complain. The same can be done for endpoint. Probably makes sense for debug only mode.

Reply:
is this considered a blocker for merging this PR?

Reply:
no